### DISCUSSION TITLE
How to generate urls with https

### DISCUSSION THREAD

**Discussion by omarherri** [serversideup/docker-php#438] [Q&A] [answered]
I'm using traefik as a reverse proxy. So i'm using Let's encrypt as a provider. What i want is to activate SSL_MODE="on". The issue is it generates SSL certificate by default. So i disabled this option to fiw the issue, but all url assets in my website are generated with http not https (Im developing my app with Laravel so i know for sure this is not related to laravel). How can be sure that urls will be generated with https.

**Comment by jaydrogers** [comment]
If you're running Vite you may need to update some configs?

https://getspin.pro/docs/services/vite#getting-vite-assets-to-load-over-https-in-production

**Reply by omarherri** [reply]
No i'm not running vite. In fact assets that are served by vite are correctly served. But i'm talking about urls for example "route('routename')". this would show a link with http so i can't get my head around that.

**Reply by jaydrogers** [reply]
What is your APP_URL set to in your .env?

**Reply by omarherri** [reply, marked as answer]
It is set with Https. But somehow i figured out what may cause the problem, so if anyone is having the same issue this typically happens because Laravel doesn't automatically detect that the request is coming through HTTPS when he is behind a reverse proxy like Traefik. The reverse proxy is terminating the SSL connection, and Nginx inside the container is seeing the request as http, even though it originally came over https.

To fix the issue you should add this to your boostrap/app:

```
->withMiddleware(function (Middleware $middleware) {
    $middleware->trustProxies(
            at: '*',
            headers: Request::HEADER_X_FORWARDED_TRAEFIK,
        );
})
```

**Reply by omarherri** [reply]
Sorry forgot to point the link for documentation: [Configuring Trusted Hosts](https://laravel.com/docs/11.x/requests#configuring-trusted-hosts)

**Reply by jaydrogers** [reply]
Thank you for posting your solution!!

Weird, I've never had to do this in any of your Traefik apps before. I learned something new today 😃
